core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -81,25 +112,91 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
         if (negativeQuery == null) {
             throw new IllegalArgumentException("boosting query requires negative query to be set");
         }
-        if (negativeBoost == -1) {
-            throw new IllegalArgumentException("boosting query requires negativeBoost to be set");
+        if (negativeBoost < 0) {
+            throw new IllegalArgumentException("boosting query requires negativeBoost to be set to positive value");
         }
         builder.startObject(NAME);
         builder.field("positive");
         positiveQuery.toXContent(builder, params);
         builder.field("negative");
         negativeQuery.toXContent(builder, params);
-
         builder.field("negative_boost", negativeBoost);
-
-        if (boost != -1) {
-            builder.field("boost", boost);
-        }
+        builder.field("boost", boost);
         builder.endObject();
     }
 
     @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        if (positiveQuery == null) {
+            validationException = QueryValidationException
+                    .addValidationError("[boosting] query requires 'positive' query to be set", validationException);
+        }
+        if (negativeQuery == null) {
+            validationException = QueryValidationException
+                    .addValidationError("[boosting] query requires 'negative' query to be set", validationException);
+        }
+        if (negativeBoost < 0) {
+            validationException = QueryValidationException
+                    .addValidationError("[boosting] query requires negativeBoost to be set to positive value", validationException);
+        }
+        return validationException;
+    };
+
+    @Override
     public String queryId() {
         return NAME;
     }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        Query positive = positiveQuery.toQuery(parseContext);
+        Query negative = negativeQuery.toQuery(parseContext);
+
+        // either inner query builder returned null
+        if (positive == null || negative == null) {
+            return null;
```

Comment:
null again scares me, I do see we used to do it before, but maybe we should change that? return no match docs query instead? or match_all?